src/challengePaths/2015/CommunityService.ts

```diff
@@ -129,20 +129,29 @@ class Test {
    * @param beCertain Whether we should check council.php instead of mafia to determine whether the test is complete.
    * @returns The output of the prepare function given, or null if the test is already complete.
    */
-  run<T>(prepare: () => T, beCertain = false): T | null {
+  run<T>(
+    prepare: () => T,
+    beCertain = false,
+    maxTurns: number | null = null
+  ): T | null {
     const finishedFunction = () =>
       beCertain ? this.verifyIsDone() : this.isDone();
     if (finishedFunction()) return null;
 
     const startTime = Date.now();
     const startTurns = myTurncount();
 
+    let submitTest = true;
     try {
       return prepare();
+    } catch {
+      submitTest = false;
     } finally {
       const prediction = this.predictor();
```

Comment:
I don't fully understand why we have a finally block here. So in either case we want to log that we did (or didn't do) the test? Why are we logging regular tests the same as failed tests. Why are we returning the value of a prepare function instead of a do function? This is smelly.

Reply:
I originally wrapped the whole thing in a try...finally because it was paralleling the various withX functions.

I think you're right that this is smelly, but `this.do()` just submits the test. I think making this return a boolean instead of the generic typing does make more sense, though.